pkg/util/azureclient/azuresdk/armmsi/user_assigned_identities.go

```diff
@@ -0,0 +1,34 @@
+package armmsi
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/msi/armmsi"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/azuresdk/azcore"
+)
+
+type UserAssignedIdentitiesClient interface {
+	Get(ctx context.Context, resourceGroupName string, resourceName string, options *armmsi.UserAssignedIdentitiesClientGetOptions) (armmsi.UserAssignedIdentitiesClientGetResponse, error)
+}
+
+type ArmUserAssignedIdentitiesClient struct {
+	*armmsi.UserAssignedIdentitiesClient
+}
+
+var _ UserAssignedIdentitiesClient = &ArmUserAssignedIdentitiesClient{}
+
+// NewUserAssignedIdentitiesClient creates a new UserAssignedIdentitiesClient
+func NewUserAssignedIdentitiesClient(subscriptionID string, credential azcore.TokenCredential, options *arm.ClientOptions) (UserAssignedIdentitiesClient, error) {
```

Comment:
Shouldn't we be returning the struct `ArmUserAssignedIdentitiesClient` here?

Reply:
Why, do we need the struct reference explicitly? All consuming code will only need to invoke the methods defined on the interface so the interface should be sufficient.

Reply:
So, the conversation around returning the non-interface started from here:- https://github.com/Azure/ARO-RP/pull/3619#discussion_r1670895639

Reply:
I'll make the change, I was unfamiliar with that Go convention.

Reply:
Done.